trace/dawr.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2022 IBM
+# Author: Akanksha J N <akanksha@linux.vnet.ibm.com>
+
+import time
+import sys
+import os
+import shutil
+import pexpect
+import tempfile
+from avocado import Test
+from avocado.utils import build, distro
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+class Dawr(Test):
+    """
+    Reading single Dawr register and multiple Dawr registers
+    with gdb interface
+    """
+
+    def setUp(self):
+        '''
+        Install the basic packages to support gdb
+        '''
+        # Check for basic utilities
+        smm = SoftwareManager()
+        self.detected_distro = distro.detect()
+        self.distro_name = self.detected_distro.name
+        deps = ['gcc', 'make', 'gdb']
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        shutil.copyfile(self.get_data('wp1.c'),
+                        os.path.join(self.teststmpdir, 'wp1.c'))
+        shutil.copyfile(self.get_data('wp2.c'),
+                        os.path.join(self.teststmpdir, 'wp2.c'))
+        shutil.copyfile(self.get_data('wp3.c'),
+                        os.path.join(self.teststmpdir, 'wp3.c'))
+        shutil.copyfile(self.get_data('Makefile'),
+                        os.path.join(self.teststmpdir, 'Makefile'))
+        build.make(self.teststmpdir)
+        os.chdir(self.teststmpdir)
+        self.temp_file = tempfile.NamedTemporaryFile().name
+
+    def test_read_dawr_v1(self):
+        """
+        Setting Read/Write watchpoint on single variable using awatch and
+        executing the program
+        """
+        child = pexpect.spawn('gdb ./wp1', encoding='utf-8')
+        time.sleep(0.3)
+        child.logfile = sys.stdout
+        child.expect('(gdb)')
+        return_value = []
+        child.sendline('awatch a')
```

Comment:
Initial few lines are common across 3 tests. May be move them to a common function??